modules/admin-ui/src/main/java/org/opencastproject/adminui/endpoint/ToolsEndpoint.java

```diff
@@ -437,26 +441,65 @@ public Response getVideoEditor(@PathParam("mediapackageid") final String mediaPa
     final MediaPackageElementFlavor sourceTracksFlavor = new MediaPackageElementFlavor("*",
       adminUIConfiguration.getThumbnailSourceFlavorSubtype());
 
-    final List<JValue> sourceTracks = Arrays.stream(mp.getTracks(sourceTracksFlavor))
-          .map(MediaPackageElement::getFlavor)
-          .map(e -> {
-            String side;
-            if (e.equals(this.adminUIConfiguration.getSourceTrackLeftFlavor())) {
-              side = "left";
-            } else if (e.equals(this.adminUIConfiguration.getSourceTrackRightFlavor())) {
-              side = "right";
-            } else {
-              side = "unknown";
-            }
-            return obj(f("flavor", obj(f("type" ,e.getType()), f("subtype", e.getSubtype()))), f("side", side));
-          })
-          .collect(Collectors.toList());
+    final Map<String, String> latestWfProperties = WorkflowPropertiesUtil
+      .getLatestWorkflowProperties(assetManager, mediaPackageId);
+    // The properties have the format "hide_flavor_audio" or "hide_flavor_video", where flavor is preconfigured.
+    // We filter all the properties that have this format, and then those which have values "true".
+    final Collection<Tuple<String, String>> hiddens = latestWfProperties.entrySet()
+      .stream()
+      .map(p -> Tuple.tuple(p.getKey().split("_"), p.getValue()))
```

Comment:
Wouldn't it make more sense to do a string comparison for the `hide_flavor` prefix?

Reply:
Well, it'll be a little shorter, but I don't think it makes _more_ sense. You'd still have to split the string to get `audio`/`video` and stuff. Or hard-code all the tests altogether.

Reply:
Ok, fair enough.  On first read I had some minor issue figuring out both the syntax and the goal.  Now that I get the syntax it's easier to read :)